notebook/notebookapp.py

```diff
@@ -831,6 +833,29 @@ def _token_changed(self, change):
         """
     )
 
+    allow_remote_access = Bool(False, config=True,
```

Comment:
allow_remote_access should be set to True when binding to anything other than localhost, otherwise we're over-complicating hosted notebook servers (e.g. this would break mybinder.org).